src/math/lp/int_solver.cpp

```diff
@@ -365,8 +365,6 @@ bool int_solver::get_freedom_interval_for_column(unsigned j, bool & inf_l, impq
             m = lcm(m, denominator(a));
 
         if (!inf_l && !inf_u) {
-            if (l > u)
```

Comment:
why are you removing this?

Reply:
It cannot happen if the column is feasible. And we assume here that it is feasible. We have l <= 0 <= u during this loop. Then we adjust l and u by shifting to xj.

Reply:
It seems to me that you are baking in some assumptions about the calling context. Maybe the current calling context is that the tableau is feasible, but this code would be easier to maintain if it doesn't rely on strong assumptions.

Reply:
The calculation will be incorrect if at least one of the basic columns 'i ' is infeasible. The feasibility assumption is already backed in to the code as I understand it.

Reply:
Actually, the code seems to be correct even when some i is infeasible. Going to reverse the change related to get_freedom...

Reply:
From the other side, if we consider the infeasible case, we cannot just "continue" the loop on l==u, expecting that further on we might encounter another infeasible variable that will create l < u!

Reply:
It looks like we have some optimizations for the feasible case that are not available in general.

Reply:
at minimum add an assertion to enforce whatever contract you assume